src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -704,25 +660,33 @@ private static void FlattenDeconstructVariables(ArrayBuilder<DeconstructionVaria
                             Error(diagnostics, ErrorCode.ERR_DeconstructionVarFormDisallowsSpecificType, component.Designation);
                         }
 
-                        return BindDeconstructionDeclarationVariables(declType, component.Designation, diagnostics);
+                        return BindDeconstructionVariables(declType, component.Designation, diagnostics);
                     }
                 case SyntaxKind.TupleExpression:
                     {
                         var component = (TupleExpressionSyntax)node;
                         var builder = ArrayBuilder<DeconstructionVariable>.GetInstance(component.Arguments.Count);
                         foreach (var arg in component.Arguments)
                         {
-                            builder.Add(BindDeconstructionDeclarationVariables(arg.Expression, diagnostics));
+                            builder.Add(BindDeconstructionVariables(arg.Expression, isDeclaration, diagnostics));
                         }
 
                         return new DeconstructionVariable(builder, node);
                     }
                 default:
-                    throw ExceptionUtilities.UnexpectedValue(node.Kind());
+                    {
+                        var boundVariable = BindExpression(node, diagnostics, invoked: false, indexed: false);
+                        if (isDeclaration && boundVariable.Kind != BoundKind.DiscardedExpression )
```

Comment:
Yes, BindIdentifier will detect that the identifier is in a context where it could either be a local or a discard. BindExpression calls BindExpression.